internal/tests/utils/utils.go

```diff
@@ -0,0 +1,21 @@
+package utils
+
+import (
+	"reflect"
+)
+
+func DeReference(in interface{}) interface{} {
+	return reflect.Indirect(reflect.ValueOf(in)).Interface()
+}
+
+func Reference(val interface{}) interface{} {
+	out := reflect.New(reflect.TypeOf(val))
+	out.Elem().Set(reflect.ValueOf(val))
+	return out.Interface()
+}
+
+func EqualTypes(in1, in2 interface{}) bool {
```

Comment:
Since `reflect.Type` is comparable, please replace it with `reflect.TypeOf(val1) == reflect.TypeOf(val2)`